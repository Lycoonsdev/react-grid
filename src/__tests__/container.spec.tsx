import { render, screen } from '@testing-library/react'

import Container from '../container'

describe('<Container /> Component', () => {
  test('render <Container />', () => {
    render(
      <Container>
        <h1>Container component</h1>
      </Container>
    )

    const container = screen.getByTestId('container')
    expect(container).toBeInTheDocument()
    expect(container).toHaveTextContent('Container component')
    expect(container).toHaveClass('container')
    expect(container).toMatchSnapshot()
  })

  test('render <Container /> with fluid props', () => {
    render(
      <Container fluid>
        <h1>Container fluid component</h1>
      </Container>
    )
    const container = screen.getByTestId('container')
    expect(container).toBeInTheDocument()
    expect(container).toHaveTextContent('Container fluid componen')
    expect(container).toHaveClass('container-fluid')
    expect(container).not.toHaveClass('container')
    expect(container).toMatchSnapshot()
  })

  test('render <Container /> with breakpoints props', () => {
    render(
      <Container sm md lg xl xxl>
        <h1>Container with breakpoints</h1>
      </Container>
    )

    const container = screen.getByTestId('container')
    expect(container).toBeInTheDocument()
    expect(container).toHaveTextContent('Container with breakpoints')
    expect(container).toHaveClass('container-sm')
    expect(container).toHaveClass('container-md')
    expect(container).toHaveClass('container-lg')
    expect(container).toHaveClass('container-xl')
    expect(container).toHaveClass('container-xxl')
    expect(container).toMatchSnapshot()
  })
})
