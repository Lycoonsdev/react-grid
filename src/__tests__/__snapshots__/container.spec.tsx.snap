// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> Component render <Container /> 1`] = `
<div
  class="container"
  data-testid="container"
>
  <h1>
    Container component
  </h1>
</div>
`;

exports[`<Container /> Component render <Container /> with breakpoints props 1`] = `
<div
  class="container container-sm container-md container-lg container-xl container-xxl"
  data-testid="container"
>
  <h1>
    Container with breakpoints
  </h1>
</div>
`;

exports[`<Container /> Component render <Container /> with fluid props 1`] = `
<div
  class="container-fluid"
  data-testid="container"
>
  <h1>
    Container fluid component
  </h1>
</div>
`;
